packages/analytics/analytics-chart/src/components/TopNTable.vue

```diff
@@ -0,0 +1,268 @@
+      v-if="description"
+      #actions
+    >
+      <div
+        class="top-n-card-description"
+        data-testid="top-n-card-description"
+      >
+        {{ description }}
+      </div>
+    </template>
+
+    <template #body>
+      <KEmptyState
+        v-if="errorMessage"
+        cta-is-hidden
+        data-testid="top-n-error-state"
+        is-error
+      >
+        <template #title>
+          {{ i18n.t('topNTable.defaultErrorStateTitle') }}
+        </template>
+        <template #message>
+          {{ errorMessage }}
+        </template>
+      </KEmptyState>
+
+      <KEmptyState
+        v-else-if="!hasData"
+        cta-is-hidden
+        data-testid="top-n-empty-state"
+        icon="stateNoData"
+        icon-size="80"
+      >
+        <template #title>
+          {{ emptyStateTitle || i18n.t('topNTable.defaultEmptyStateTitle') }}
+        </template>
+      </KEmptyState>
+
+      <div
+        v-else
+        class="top-n-table"
+        data-testid="top-n-table"
+      >
+        <div class="table-headings">
+          <div class="table-row">
+            <div class="column-1">
+              {{ i18n.t('topNTable.nameLabel') }}
+            </div>
+            <div class="column-2">
+              {{ columnName }}
+            </div>
+          </div>
+        </div>
+
+        <div
+          class="table-body"
+          data-testid="top-n-data"
+        >
+          <div
+            v-for="(entry, idx) in records"
+            :key="`entry-${idx}`"
+            class="table-row"
+            :data-testid="`row-${getId(entry)}`"
+          >
+            <div class="column-1">
+              <slot
+                name="name"
+                :record="{
+                  id: getId(entry),
+                  name: getName(entry),
+                  dimension: displayKey,
+                }"
+              >
+                {{ getName(entry) }}
+              </slot>
+            </div>
+            <div class="column-2">
+              &nbsp; {{ getValue(entry) }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </template>
+  </KCard>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import { AnalyticsExploreV3Result, AnalyticsExploreV3Meta, AnalyticsExploreRecord, RecordEvent } from '@kong-ui-public/analytics-utilities'
+// @ts-ignore - approximate-number no exported module
+import approxNum from 'approximate-number'
+import composables from '../composables'
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    default: '',
+  },
+  data: {
+    type: Object as PropType<AnalyticsExploreV3Result>,
+    required: true,
+  },
+  emptyStateTitle: {
+    type: String,
+    default: '',
+  },
+})
+
+const { i18n } = composables.useI18n()
+const meta = computed((): AnalyticsExploreV3Meta => props.data.meta)
+const records = computed((): AnalyticsExploreRecord[] => props.data.records)
```

Comment:
question: why are these props set to computed properties? The props are already reactive if you just reference them below as needed

Reply:
They are computeds just in an attempt to simplify the reuse below so I can say `meta.value` instead of `props.data.meta`. If there is a drawback to setting them as computeds I can remove.

Reply:
I'd recommend just referencing the props directly; there's no real point in making Vue go through extra cycles since it's 1:1